topostats/tracing/dnatracing.py

```diff
@@ -1462,6 +1462,9 @@ def analyse_nodes(self, max_branch_length: float = 20e-9):
                                 self.image, single_branch_img, single_branch_coords, [x, y]
                             )  # hess_area
                             matched_branches[i]["avg_mask"] = mask
+                            avg_img = avg_img[
```

Comment:
This looks good to me, but `image_slices` are currently initialised later on in the code (line 1548 in dnatracing.py) and so the code currently raises `UnboundLocalError: local variable 'image_slices' referenced before assignment`. This should be resolved by initialising `image_slices` ahead of its first use in line 1465!

Reply:
Ah yep my bad, fixed by moving `image_slices` to the start of the node iteration loop